CMakeLists.txt

```
@@ -111,6 +111,8 @@ set(debug_flags_to_test
 if(MSVC)
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /NXCompat /DynamicBase")
     set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /NXCompat /DynamicBase")
+else()
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -lm")
```

```suggestion
    find_library(M_LIB name m)
    if(M_LIB)
        link_libraries(m)
    endif()
```
Although project wide linking is highly not recommended and it is prefered to use target_link_libraries on the specific library or executable. If you could mention what actually needs libm, I could help recommend something else (I think you want it in Codec?)